include/pisa/topk_queue.hpp

```diff
@@ -10,7 +10,7 @@ using Threshold = float;
 struct topk_queue {
     using entry_type = std::pair<float, uint64_t>;
```

Comment:
This is not related to the issue but it just occurred to me: there is no reason for the entry to hold `uint64_t`, is there? Notice that because of this, there will be some padding after the float and the entire entry is now 16 bytes when it could be just 8 (we use 4 byte integers for documents elsewhere anyway, so we don't support any bigger anyway). This is of course a relatively small structure but I'm thinking that cutting the size in half could maybe improve caching. And I see no downsides, either. So I suggest changing it to `std::pair<float, uint32_t>`.

Reply:
Then, we should change the type of `k` as well for good measure, to be consistent.

Reply:
Good point. We should have a type definition of DocID somewhere and use always that one.
We can do it in a separate PR if you prefer.

Reply:
We might as well change it to `uint32_t` here, but since the alias will touch a lot of files, maybe that part should be done separately.

Reply:
Agreed, but yes, let's do this in a separate PR.